addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3380,9 +3380,21 @@ export class OdooEditor extends EventTarget {
         if (!rangeContent.hasChildNodes()) {
             return;
         }
+        // find list of ancestors.
+        const ancestorList = range.commonAncestorContainer.nodeType === Node.ELEMENT_NODE ?
+                            ancestors(range.commonAncestorContainer.firstChild) : ancestors(range.commonAncestorContainer);
+        const closestLi = closestElement(selection.anchorNode, 'li');
+        for (const ancestor of ancestorList) {
+            if ((closestLi && rangeContent.firstChild.nodeName === 'LI') || isBlock(rangeContent.firstChild)) {
+                break;
+            }
+            const clone = ancestor.cloneNode();
+            clone.append(...rangeContent.childNodes);
+            rangeContent.appendChild(clone);
+        }
         // Repair the copied range.
         if (rangeContent.firstChild.nodeName === 'LI') {
-            const list = range.commonAncestorContainer.cloneNode();
+            const list = closestLi.parentElement.cloneNode();
```

Comment:
I suppose this is related to the special case with `closestLi` and the `break` above, but since I don't understand the special case, I don't understand the change here either. Maybe I will understand it when you have added a comment.